tracing/opencensus/endpoint_options.go

```diff
@@ -11,6 +15,16 @@ type EndpointOptions struct {
 	// Attributes holds the default attributes which will be set on span
 	// creation by our Endpoint middleware.
 	Attributes []trace.Attribute
+
+	// GetName holds the function used for generating the span name
+	// based on the current name and from the information found in the incoming Request.
+	//
+	// If the returned name is empty, the existing name for the endpoint is kept.
+	GetName func(ctx context.Context, name string) string
+
+	// GetAttributes holds the function used for extracting additional attributes
+	// from the information found in the incoming Request.
+	GetAttributes func(ctx context.Context) []trace.Attribute
```

Comment:
I believe attributes are just attached to the span(s), is that correct? Are these attributes used instead of, or in addition to, the Attributes field?

Reply:
Yes, they are attached, in addition to any other attributes.

Reply:
> GetAttributes is an optional function that can extract trace attributes from the context and add them to the span.